monai/handlers/roc_auc.py

```diff
@@ -96,6 +98,17 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
     def compute(self):
         _prediction_tensor = torch.cat(self._predictions, dim=0)
         _target_tensor = torch.cat(self._targets, dim=0)
+
+        if dist.is_available() and dist.is_initialized() and not self._is_reduced:
+            # create placeholder to collect the data from all processes:
+            output = [torch.zeros_like(_prediction_tensor) for _ in range(dist.get_world_size())]
+            dist.all_gather(output, _prediction_tensor)
+            _prediction_tensor = torch.cat(output, dim=0)
+            output = [torch.zeros_like(_target_tensor) for _ in range(dist.get_world_size())]
+            dist.all_gather(output, _target_tensor)
+            _target_tensor = torch.cat(output, dim=0)
+            self._is_reduced = True
+
         return compute_roc_auc(
```

Comment:
Here, either `compute_roc_auc` should know how to deal with tensors on CUDA (if nccl dist backend), either we need to put them to the CPU before. And here, all processess will execute on the CPU the same computation which maybe a bit redundant. That's why I talked about broadcasting in ignite's issue.

Reply:
`compute_roc_auc` can handle CUDA tensor here.
As you said in ignite thread, maybe `all_gather` is compatible for all kinds of backends, I think we can use this logic for the first version.
Thanks.

Reply:
> As you said in ignite thread, maybe all_gather is compatible for all kinds of backends, I think we can use this logic for the first version.

Just to detail that: `all_gather`/`gather` works for all torch native backends. In ignite thread (just added it), I was talking about ignite supported backend (torch native, xla, horovod). So, Horovod and XLA do not have `gather` collective operation, that's why I proposed to keep `all_gather`.

Reply:
Thanks for your explanation!